### DISCUSSION TITLE
Is it possible to embed an iframe?

### DISCUSSION THREAD

**Discussion by fonzane** [sibiraj-s/ngx-editor#422] [Q&A] [answered]
Hi,

I was wondering if it is possible to create a button in the editor which embeds an iframe, similar to an image? Similar to the image button, I just want to provide an url and a title and then embed this into an iframe tag. Is it possible to create such a button myself?

Kind regards

**Comment by sibiraj-s** [comment, marked as answer]
Yes. You can follow the resources

1. See https://sibiraj-s.github.io/ngx-editor/#/menu?id=custom-menu to insert custom element into the editors menubar.
2. Replace Image component implementation to add your own function with iframe
   - https://github.com/sibiraj-s/ngx-editor/tree/master/projects/ngx-editor/src/lib/components/image-view
   - https://github.com/sibiraj-s/ngx-editor/blob/master/projects/ngx-editor/src/lib/plugins/image-resize.ts (optional source)
3. Create a schema to let the editor know how to parse/handle your element https://github.com/sibiraj-s/ngx-editor/blob/master/projects/ngx-editor/schema/nodes.ts#L205

**Reply by fonzane** [reply]
Thanks for your answer. In order to modify the state of the editor and insert the iframe element, it seems I have to install the 'prosemirror-state' package. Is that correct? Or can I import these packages somewhere from ngx-editor?

**Reply by sibiraj-s** [reply]
Yes. The editor itself is built on top of prosemirror. You are more or less creating a prosemirror plugin and making it work using angular components

**Comment by jorgemddev** [comment]
Hello, could you create an example of how to insert a custom element, it would be very useful

**Reply by sibiraj-s** [reply]
HI @jorgemddev 

See https://sibiraj-s.github.io/ngx-editor/en/examples/codemirror/. does this help?

**Reply by jorgemddev** [reply]
thanks for your time

[a comment by an automated account is not shown]
